dlib/test/serialize.cpp

```diff
@@ -401,18 +401,52 @@ namespace
         dlib::deserialize(item.b_false,in);
     }
     
+    template<typename T>
+    bool pointers_equal(const unique_ptr<T>& l, const unique_ptr<T>& r)
+    {
+        bool ptrs_equal = l.operator bool() == r.operator bool();
+        if (ptrs_equal && l)
+            ptrs_equal &= (*l == *r);
+        return ptrs_equal;
+    }
+
+    template<typename T>
+    bool pointers_equal(const shared_ptr<T>& l, const shared_ptr<T>& r)
+    {
+        bool ptrs_equal = l.operator bool() == r.operator bool();
+        if (ptrs_equal && l)
+            ptrs_equal &= (*l == *r);
+        return ptrs_equal;
```

Comment:
same thing here

Reply:
Again, ```l == r```, is the same as ```l.get() == r.get()``` i think. We don't care if the raw underlying pointers are equal.